metaflow/client/core.py

```diff
@@ -1024,16 +1024,62 @@ def environment_info(self):
         env = [m for m in ENVIRONMENTS + [MetaflowEnvironment] if m.TYPE == env_type][0]
         return env.get_client_info(self.path_components[0], self.metadata_dict)
 
-    def _load_log(self, logtype, as_unicode=True):
+    def _load_log(self, stream):
+        log_location = self.metadata_dict.get('log_location_%s' % stream)
+        if log_location:
+            return self._load_log_legacy(log_location)
+        else:
+            return ''.join(line + '\n' for _, line in self.loglines(stream))
+
+    def loglines(self, stream, as_unicode=True):
+        """
+        Return an iterator over (utc_timestamp, logline) tuples.
+
+        If as_unicode=False, logline is returned as a byte object. Otherwise,
+        it is returned as a (unicode) string.
+        """
+        from metaflow.mflog.mflog import merge_logs
+        from metaflow.mflog import LOG_SOURCES
+        from metaflow.datastore import DATASTORES
```

Comment:
why are those imports here? recursive dep? may be worth adding a comment

Reply:
No recursive dep. They are only used in this function. I can also move them at the top.

Reply:
Yep I think it would be better to have them at the top then. Otherwise makes one think there is something special going on here :)